### PR TITLE
Add exception filter

### LINKED ISSUE #35: Add exception filter using nest js

### PR DESCRIPTION
**Pull request by erezmus**
closes #35

### CODE REVIEW

File: docker-compose.test.yml [hunk 1]
```diff
@@ -0,0 +1,19 @@
+## This is only a single container right now but in future this should bring up supporting services needed for development ie: postgres, audit service ect.
+version: '3'
```
Comment (norisalsaadie) [inline docker-compose.test.yml]: Does this version mean anything?

File: src/packages/submission/submission.controller.ts [hunk 2]
```diff
@@ -1,10 +1,12 @@
-import { Logger, HttpException, HttpStatus } from '@nestjs/common';
+import { Logger, HttpException, HttpStatus, UseGuards } from '@nestjs/common';
 import { Option, None } from 'funfix';
 import { SubmissionRepository, SubmissionId, Submission } from './submission.types';
+import { AuthGuard } from '@nestjs/passport';
 
 // TODO: this is temporary until the picture on validation and article types is clearer.
 const articlesTypes = ['researchArticle', 'featureArticle', 'researchAdvance'];
 
+@UseGuards(AuthGuard)
```
Comment (norisalsaadie) [inline src/packages/submission/submission.controller.ts]: This should probably be reverted since it's not related to this PR

File: Makefile [hunk 3]
```diff
@@ -14,6 +19,13 @@ lint: get_deps
 test: get_deps
 	yarn test
 
+test_integration:
```
Comment (diversemix) [inline Makefile]: This looks dependent on `setup_gitmodules` /cc @erezmus  @norisalsaadie

Comment (erezmus) [reply Makefile]: setup_gitmodules is usually done manually or as part of setup (in reviewer repo).

**Pull request merged**